### ISSUE TITLE
h-event time test should not normalise timezone -v2

### ISSUE THREAD

**Issue by kylewm** [microformats/tests#29]
in `microformats-v2/h-event/time.html`

```
<span class="h-event">
    <span class="p-name">The 4th Microformat party</span> will be on 
    <ul>
        <li class="dt-start">
            <time class="value" datetime="2009-06-26">26 July</time>, from
            <time class="value">19:00:00-08:00</time> 
        </li>
        <li class="dt-start">
            <time class="value" datetime="2009-06-26">26 July</time>, from
            <time class="value">19:00:00-0800</time> 
        </li>
...

```

the output reads

```
"start": ["2009-06-26T19:00:00-0800", "2009-06-26T19:00:00-0800"...
```

with the colon removed from the timezone. There is an issue discussing normalisation, but for now http://microformats.org/wiki/value-class-pattern#Date_and_time_parsing seems to indicate the colon should be preserved.

**Comment by glennjones** [comment]
Yes your right. My parser is been a bit too clever and normalising dates. I am using ISO 8601 date "profiles" which are like a set of sub rules that restrict the ISO 8601. The default profile drops colon from timezone.

The tests needs changing and I need to add a flag for "date normalisation" which is switch off by default. Unless of cause we add date normalisation to parsing rules : ) 

I will do this in next round of updates

**Comment by kylewm** [comment]
Normalizing is a proposal, I have it implemented and commented out in
mf2py. Happy to pursue that route instead if you prefer


On Wed, Jun 10, 2015, at 09:11 AM, Glenn Jones wrote:
> Yes your right. My parser is been a bit too clever and normalising
> dates. I am using ISO 8601 date "profiles" which are like a set of sub
> rules that restrict the ISO 8601. The default profile drops colon from
> timezone.


> The tests needs changing and I need to add a flag for "date
> normalisation" which is switch off by default. Unless of cause we add
> date normalisation to parsing rules : )


> I will do this in next round of updates


> — Reply to this email directly or view it on GitHub[1].

>


Links:

  1. https://github.com/microformats/tests/issues/29#issuecomment-110815472

**Comment by glennjones** [comment]
The tests now do NOT normalising dates as per current parsing rules. Leave this issue open while normalising dates is discussed.

**Comment by willnorris** [comment]
@tantek recently resolved http://microformats.org/wiki/microformats2-parsing-issues#parsing_a_dt-_property.  It seemed to be mainly focused on normalizing dates with regards to space vs "T" as a separator and removal of implying 00 seconds.  However, the vcp page is also now pretty clear that a resulting dt value should not have colons in the timezone.

**Comment by dissolve** [comment]
h-event/dates example is normalizing to add the colons still.

**Comment by jalcine** [comment]
Nudging this because this test is the only v2 test that's failing for the Rust library. Should we opt to keep (current behavior expected in the Rust library), preserve as authored, or drop the colon in offsets?

(Originally published at: https://jacky.wtf/2022/6/kduV)

**Comment by gRegorLove** [comment]
Specification updated (https://github.com/microformats/microformats2-parsing/issues/56) to normalize by removing colon in timezone offset, so these tests can be updated accordingly.

**Comment by gRegorLove** [comment]
I reviewed the master branch and confirmed that https://github.com/microformats/tests/commit/17d31dbeb3d64426d5a15548ea778f61df73c28f normalized the test outputs, so no additional test updates are needed to resolve this issue.
